components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -1670,4 +1696,49 @@ public WorkflowResponse completeWorkflow(WorkflowExecutor workflowExecutor, Work
         }
         return response;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void removePendingLifecycleWorkflowTaskForAPI(String apiId) throws APIManagementException {
+        
+        API api = getApiDAO().getAPI(apiId);
+        if (api != null) {
+            if (APILCWorkflowStatus.PENDING.toString().equals(api.getWorkflowStatus())) {
+                try {
+                    //change the state back
+                    getApiDAO().updateAPIWorkflowStatus(apiId, APILCWorkflowStatus.APPROVED);
+                    
+                    // call executor's cleanup task
+                    String workflowExtRef = getWorkflowDAO().getExternalWorkflowReferenceForPendingTask(apiId,
+                            WorkflowConstants.WF_TYPE_AM_API_STATE);
+                    if (workflowExtRef != null) {
+                        WorkflowExecutor executor = WorkflowExecutorFactory.getInstance()
+                                .getWorkflowExecutor(WorkflowConstants.WF_TYPE_AM_API_STATE);
+                        try {
+                            executor.cleanUpPendingTask(workflowExtRef);
+                        } catch (WorkflowException e) {
+                            String warn = "Failed to clean pending api state change task for " + apiId;
+                            // failed cleanup processes are ignored to prevent failing the deletion process
+                            log.warn(warn, e.getLocalizedMessage());
+                        }
+                        getWorkflowDAO().deleteWorkflowEntryforExternalReference(workflowExtRef);
+                    }
+                    
+                } catch (APIMgtDAOException e) {
+                    String msg = "Error occurred while changing api lifecycle workflow status";
+                    log.error(msg, e);
+                    throw new APIManagementException(msg, ExceptionCodes.APIMGT_DAO_EXCEPTION);
+                }
+            } else {
+                String msg = "API does not have a pending lifecycle state change.";
+                log.error(msg);
+                throw new APIManagementException(msg, ExceptionCodes.WORKFLOW_NO_PENDING_TASK);
+            }
+        } else {
+            log.error("Couldn't found API with ID " + apiId);
+            throw new APIManagementException("Couldn't find API with ID " + apiId, ExceptionCodes.API_NOT_FOUND);
```

Comment:
Better to assign [1] to variable. used twice
[1] Couldn't find API with ID " + apiId